### ISSUE TITLE
Catalog set to null causes error with MySQL driver.

### ISSUE THREAD

**Issue by BrassyPanache** [brettwooldridge/HikariCP#500]
When creating a new database (e.g. with Flyway) it is often the case that you want to connect with the default catalog for a given driver.  In the Flyway instance it seems that the catalog is set to the new database after the connection is established.  If the catalog has been changed and the initial value was unset, this exception is thrown:

```java
17:37:49.623 [main] ERROR o.f.c.internal.util.jdbc.JdbcUtils - Error while closing Jdbc connection
java.sql.SQLException: Catalog can not be null
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:959) ~[mysql-connector-java-5.1.37.jar:5.1.37]
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:898) ~[mysql-connector-java-5.1.37.jar:5.1.37]
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:887) ~[mysql-connector-java-5.1.37.jar:5.1.37]
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:862) ~[mysql-connector-java-5.1.37.jar:5.1.37]
	at com.mysql.jdbc.ConnectionImpl.setCatalog(ConnectionImpl.java:4913) ~[mysql-connector-java-5.1.37.jar:5.1.37]
	at com.zaxxer.hikari.pool.PoolElf.resetConnectionState(PoolElf.java:263) ~[HikariCP-2.4.1.jar:na]
	at com.zaxxer.hikari.pool.PoolBagEntry.resetConnectionState(PoolBagEntry.java:122) ~[HikariCP-2.4.1.jar:na]
	at com.zaxxer.hikari.proxy.ConnectionProxy.close(ConnectionProxy.java:193) ~[HikariCP-2.4.1.jar:na]
	at org.flywaydb.core.internal.util.jdbc.JdbcUtils.closeConnection(JdbcUtils.java:71) ~[flyway-core-3.2.1.jar:na]
	at org.flywaydb.core.Flyway.execute(Flyway.java:1424) [flyway-core-3.2.1.jar:na]
	at org.flywaydb.core.Flyway.migrate(Flyway.java:1006) [flyway-core-3.2.1.jar:na]
```

I feel like the fix would be to simply never update the catalog with a null value.  Perhaps use the same logic if the catalog is never set in the first place.  The code below allows a null update if the currentCatalog is non-null and the new catalog is null.

```java
final String currentCatalog = poolEntry.catalog;
if ((currentCatalog != null && !currentCatalog.equals(catalog)) || (currentCatalog == null && catalog != null)) {
     poolEntry.connection.setCatalog(catalog);
     resetBits |= 0b01000;
}
```

**Comment by nitincchauhan** [comment]
If this is similar to https://github.com/brettwooldridge/HikariCP/issues/410, It is fixed in 2.4.2

**Comment by BrassyPanache** [comment]
Verified as fixed in 2.4.2.  Thanks!
